packages/components/src/drop-zone/index.js

```diff
@@ -7,13 +7,41 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component, createRef } from '@wordpress/element';
+import { useContext, useEffect, useState, useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Dashicon from '../dashicon';
-import { DropZoneConsumer } from './provider';
+import { DropZoneConsumer, Context } from './provider';
+
+export function useDropZone( { element, onFilesDrop, onHTMLDrop, onDrop, isDisabled } ) {
```

Comment:
Should we add some JSDocs and types :p cc @aduth

Reply:
Should we move this to `@wordpress/compose` like all other hooks? I guess the problem is that it's dependent on the DropzoneProvider so I'm on the fence. I know the provider is necessary for performance reasons, other than that it would be good to kill.

Reply:
I haven't looked into the `DropzoneProvider`, but as long as we need that, this seems like the right place. We can prefix it with `__unstable` if we want to keep our options open.

Reply:
I prefixed it. I will have another look separately to move it and look into the provider.